### PR TITLE
add redis config

### PR DESCRIPTION
**Pull request by moosecanswim**
Add in configuration to operator to specify the redis url to use.

### CODE REVIEW

**Comment by moosecanswim** [comment]
[CH 6706]

File: api/v1alpha1/operatorconfig_types.go [hunk 1]
```diff
@@ -31,7 +31,16 @@ type OperatorConfig struct {
...
+	ImagePullSecretName string      `json:"imagePullSecretName"`
+	RedisConfig         redisConfig `json:"redis"`
+	// RedisUrl                   string `json:"redis.url"`
+	// RedisCertificateSecretName string `json:"redis.certificateSecretName"`
+
+}
+
+type redisConfig struct {
+	Url        string `json:"url"`
+	SecretName string `json:"certificateSecretName"`
```
Comment (bcmendoza) [inline api/v1alpha1/operatorconfig_types.go:43]: Can you move this into the `store` package and export it?

File: main.go [hunk 2]
```diff
@@ -53,13 +53,18 @@ func main() {
 	var probeAddr string
 	var enableLeaderElection bool
 	var development bool
+	var redisUrl string
```
Comment (bcmendoza) [inline main.go:56]: We don't need command line args for each option -- just the ones the controller-runtime manager requires, plus a `development` flag for convenience in local development. You can remove lines 56-57 and 65-66.

File: main.go [hunk 3]
```diff
@@ -97,6 +102,17 @@ func main() {
...
+		opConfig.RedisConfig.Url = "redis://greymatteroperator:redis@localhost:6379/0" // TODO: if this field is empty and we will be spinning up a redis instance then it should connect to it <svc name>.<current namespace>.svc.cluster.local
+	}
+	// TODO: if certificateSecretName exists then use tls to connect to redis (https://redis.uptrace.dev/guide/server.html#using-tls)
+	if opConfig.RedisConfig.SecretName != "" {
+		// get secret specified
+		// validate there are 'ca', 'server_cert', 'server_key' keys.
+		// set up redis to use tls
+		setupLog.Info("A redis config secret name was provided.  TODO: Need to set up redis tls")
+	}
+
```
Comment (bcmendoza) [inline main.go:115]: Let's save this initialization logic for when we call `store.New()`. Maybe move this into `store.go` in a separate function for now?

**Review by bcmendoza — CHANGES_REQUESTED** [review]

File: main.go (hunk 3 above)
Comment (moosecanswim) [reply main.go]: done.  also added testing.

**Pull request closed without merging**